Client/Frontend/Browser/Playlist/PlaylistCarplayManager.swift

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Combine
+import MediaPlayer
+import Shared
+import Data
+
+private let log = Logger.browserLogger
+
+/// Lightweight class that manages a single MediaPlayer item
+/// The MediaPlayer is then passed to any controller that needs to use it.
+class PlaylistCarplayManager: NSObject {
+    private var carPlayStatusObservers = [Any]()
+    private var contentManager = MPPlayableContentManager.shared()
+    private var carPlayController: PlaylistCarplayController?
+    private weak var mediaPlayer: MediaPlayer?
+    private(set) var isCarPlayAvailable: Bool = false
+    
+    var currentlyPlayingItemIndex: Int = -1
+    var currentPlaylistItem: PlaylistInfo?
+    
+    // There can only ever be one instance of this class
+    // Because there can only be a single AudioSession and MediaPlayer
+    // in use at any given moment
+    static let shared = {
+        PlaylistCarplayManager()
```

Comment:
why a closure is needed here? can we just simply write `static let shared = PlaylistCarplayManager()`